bluetooth: controller: radio: Use HAL API in radio_nrf5_ppi

auroraslb opened the pull request:
Rewrite some functions in radio_nrf5_ppi to use available HAL APIs for subscribe and publish. This will make it possible to test the controller in a simulator. Also add guards on TXPOWER to ensure positive values only are used if they are defined.

auroraslb committed:
bluetooth: controller: radio: Use HAL API in radio_nrf5_ppi

Rewrite to use subscribe and publish functions for the
different peripherals. This will make it possible to test
the controller in a simulator.

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>

cvinayak reviewed (changes requested):
Good work.

Some change requests:
1. Make a separate PR for the TX power commit, so it can be approved independently.
2. Add a new commit, which moves the nRF53 DPPI to radio_nrf5_dppi.h (related to https://github.com/zephyrproject-rtos/zephyr/issues/26118)

auroraslb commented:
> Good work.
> 
> Some change requests:
> 
>     1. Make a separate PR for the TX power commit, so it can be approved independently.
> 
>     2. Add a new commit, which moves the nRF53 DPPI to radio_nrf5_dppi.h (related to #26118)

Hi, I've created a separate PR for the TX power changes (#30880).
I will look more into a new commit, which moves the nRF53 DPPI to radio_nrf5_dppi.h.

cvinayak commented:
@auroraslb FYI, I have cherry-picked your changes with some changes here #30950, do review and take the changes as necessary/appropriate into this PR.

auroraslb committed:
bluetooth: controller: radio: Split radio_nrf5_ppi.h

Splits the file into radio_nrf5_ppi.h and radio_nrf5_dppi.h
and moves the nRF53 DPPI to radio_nrf5_dppi.h.

Signed-off-by: Aurora Sletnes Bjørlo <aurora.sletnes@nordicsemi.no>

carlescufi commented:
@auroraslb will you follow-up on this PR?

cvinayak commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5.h:
```diff
@@ -40,5 +40,10 @@
 
+#if defined(PPI_PRESENT)
 #include "radio_nrf5_ppi.h"
+#endif
+#if defined(DPPI_PRESENT)
+#include "radio_nrf5_dppi.h"
+#endif
```
```suggestion
#elif defined(DPPI_PRESENT)
#include "radio_nrf5_dppi.h"
#else
#error "PPI or DPPI abstractions missing."
#endif
```

cvinayak reviewed (changes requested):
I see that my original change requests have been addressed. Makes sense that this PR gets merged prior to #31336. I can rebase #31336 thereafter.

rugeGerritsen committed:
bluetooth: controller: radio: Fail if PPI/DPPI header not present

The hal requires one of these to be present.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

cvinayak commented on subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5.h:
```diff
@@ -40,5 +40,14 @@
+
+#if defined(PPI_PRESENT)
 #include "radio_nrf5_ppi.h"
+#endif
+#if defined(DPPI_PRESENT)
```
```suggestion
#elif defined(DPPI_PRESENT)
```

cvinayak commented:
@rugeGerritsen any progress here? This is blocking #31336 and  #30708

rugeGerritsen commented:
@cvinayak , all work in this PR is done. I see you have approved it. It is ready to be merged.
